dandi/tests/test_files.py

```diff
@@ -0,0 +1,178 @@
+from operator import attrgetter
+from pathlib import Path
+
+from dandischema.models import get_schema_version
+import numpy as np
+import zarr
+
+from .. import get_logger
+from ..consts import ZARR_MIME_TYPE, dandiset_metadata_file
+from ..dandiapi import RemoteZarrAsset
+from ..files import (
+    DandisetMetadataFile,
+    GenericAsset,
+    NWBAsset,
+    ZarrAsset,
+    dandi_file,
+    find_dandi_files,
+)
+
+lgr = get_logger()
+
+
+def test_find_dandi_files(tmp_path: Path) -> None:
+    (tmp_path / dandiset_metadata_file).touch()
+    (tmp_path / "sample01.zarr").mkdir()
+    (tmp_path / "sample01.zarr" / "inner.nwb").touch()
```

Comment:
FWIW, I really like `@with_tree` we have in DataLad: https://github.com/datalad/datalad/blob/master/datalad/tests/utils.py#L543 which would take care also about populating directories, files or even `.tar.gz` archives.  Might consider adopting/creating similar one for dandi-cli